### PR TITLE
Add tchannel server and client

### PR DESCRIPTION
**Pull request by ChuntaoLu**
This PR adds tchannel server and client implementation. 

~I will soon add~ **Update**, following has been added:
- meta handler for health and version info
- integration tests 

r: @Raynos @Matt-Esch @zhenghuiwang

### CODE REVIEW

File: runtime/tchannel/struct.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tchannel
+
+import (
+	"bytes"
+	"io"
+
+	"code.uber.internal/rt/tchan-server/thriftrw/buffer"
```
Comment (Raynos) [inline runtime/tchannel/struct.go]: cannot reference code.uber.internal ...

Comment (ChuntaoLu) [inline runtime/tchannel/struct.go]: oops, good catch

Comment (ChuntaoLu) [inline runtime/tchannel/struct.go]: this should have been stashed, not sure how it went in...

Comment (Raynos) [inline runtime/tchannel/struct.go]: text editor adds imports on save and sometimes it "infers" the wrong thing to import.

**Comment by Raynos** [comment]
This is a lot for one PR. 

Let's move all the meta related code to a separate PR.

Feel free to use stacked PRs

**Pull request closed without merging**